lib/msal-browser/src/crypto/BrowserCrypto.ts

```diff
@@ -28,36 +30,85 @@ const PUBLIC_EXPONENT: Uint8Array = new Uint8Array([0x01, 0x00, 0x01]);
  */
 export class BrowserCrypto {
 
-    private _keygenAlgorithmOptions: RsaHashedKeyGenParams;
+    private keygenAlgorithmOptions: RsaHashedKeyGenParams;
     private subtleCrypto: ISubtleCrypto;
+    private logger: Logger;
+    private cryptoOptions?: CryptoOptions;
 
-    constructor() {
-        if ("crypto" in window) {
+    constructor(logger: Logger, cryptoOptions?: CryptoOptions) {
+        this.logger = logger;
+        this.cryptoOptions = cryptoOptions;
+
+        if (this.hasBrowserCrypto()) {
+            // Use standard modern web crypto if available
+            this.logger.verbose("BrowserCrypto: modern crypto interface available");
             this.subtleCrypto = new ModernBrowserCrypto();
-        } else if ("msrCrypto" in window) {
-            this.subtleCrypto = new MsrBrowserCrypto();
-        } else if ("msCrypto" in window) {
+        } else if (this.hasIECrypto()) {
+            // For IE11, use msCrypto interface
+            this.logger.verbose("BrowserCrypto: MS crypto interface available");
             this.subtleCrypto = new MsBrowserCrypto();
+        } else if (this.hasMsrCrypto() && this.cryptoOptions?.useMsrCrypto) {
+            // For other browsers, use MSR Crypto if found
+            this.logger.verbose("BrowserCrypto: MSR crypto interface available");
+            this.subtleCrypto = new MsrBrowserCrypto();
         } else {
-            throw BrowserAuthError.createCryptoNotAvailableError("Browser crypto, msCrypto, or msrCrypto objects not available.");
+            if (this.hasMsrCrypto()) {
+                this.logger.info("BrowserCrypto: MSR Crypto interface available but system.cryptoOptions.useMsrCrypto not enabled");
+            }
+            this.logger.error("BrowserCrypto: No crypto interfaces available.");
+            throw BrowserAuthError.createCryptoNotAvailableError("Browser crypto, msCrypto, or msrCrypto interfaces not available.");
+        }
+
+        // Mainly needed for MSR Crypto: https://github.com/microsoft/MSR-JavaScript-Crypto#random-number-generator-prng
+        if (this.subtleCrypto.initPrng) {
+            this.logger.verbose("BrowserCrypto: Interface requires entropy");
+
+            if (!this.cryptoOptions?.entropy) {
+                this.logger.error("BrowserCrypto: Interface requires entropy but none provided.");
+                throw BrowserConfigurationAuthError.createEntropyNotProvided();
+            }
+
+            this.logger.verbose("BrowserCrypto: Entropy provided");
+            this.subtleCrypto.initPrng(this.cryptoOptions.entropy);
         }
 
-        this._keygenAlgorithmOptions = {
+        this.keygenAlgorithmOptions = {
```

Comment:
Why the rename here?

Reply:
Moving away from underscore prefixed properties, which we don't really do anywhere else.